commit 1157c2683c5b213588c87b7e524416f78045476b
Author: Doug Smythies <doug.smythies@gmail.com>
Date:   Mon Dec 17 23:34:04 2018 -0800

    tools/power/x86/intel_pstate_tracer: Fix non root execution for post processing a trace file
    
    [ Upstream commit 663546903c835fe46308b1b1e53d32d1f2b33da9 ]
    
    This script is supposed to be allowed to run with regular user
    privileges if a previously captured trace is being post processed.
    
    Commit fbe313884d7d (tools/power/x86/intel_pstate_tracer: Free the
    trace buffer memory) introduced a bug that breaks that option.
    
    Commit 35459105deb2 (tools/power/x86/intel_pstate_tracer: Add
    optional setting of trace buffer memory allocation) moved the code
    but kept the bug.
    
    This patch fixes the issue.
    
    Fixes: 35459105deb2 (tools/power/x86/intel_pstate_tracer: Add optional ...)
    Signed-off-by: Doug Smythies <dsmythies@telus.net>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/tools/power/x86/intel_pstate_tracer/intel_pstate_tracer.py b/tools/power/x86/intel_pstate_tracer/intel_pstate_tracer.py
index 84e2b648e622..2fa3c5757bcb 100755
--- a/tools/power/x86/intel_pstate_tracer/intel_pstate_tracer.py
+++ b/tools/power/x86/intel_pstate_tracer/intel_pstate_tracer.py
@@ -585,9 +585,9 @@ current_max_cpu = 0
 
 read_trace_data(filename)
 
-clear_trace_file()
-# Free the memory
 if interval:
+    clear_trace_file()
+    # Free the memory
     free_trace_buffer()
 
 if graph_data_present == False:
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1157_1157c2683c5b213588c87b7e524416f78045476b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1157_1157c2683c5b213588c87b7e524416f78045476b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
